### PR TITLE
Identity Layer implementation

### PR DESCRIPTION
**Pull request by shubham1206agra**
Implemented Identity Layer

### CODE REVIEW

**Comment by shubham1206agra** [comment]
@zoq @rcurtin Can you please merge this?

**Review by rcurtin — COMMENTED** [review]
I know it must seem like the most ridiculous thing ever, since the layer is so incredibly simple, but do you mind adding a test for it?  You never know when later changes might accidentally break this layer... good to have a test that ensures it's behaving correctly.  (It can be a super easy test.  Just make sure the output is the same as the input.  Maybe it's also worth checking that the output is not an alias of the input.)

Otherwise everything here looks great to me. :+1:

**Comment by shubham1206agra** [comment]
> I know it must seem like the most ridiculous thing ever, since the layer is so incredibly simple, but do you mind adding a test for it? You never know when later changes might accidentally break this layer... good to have a test that ensures it's behaving correctly. (It can be a super easy test. Just make sure the output is the same as the input. Maybe it's also worth checking that the output is not an alias of the input.)
> 
> Otherwise everything here looks great to me. 👍

How do I check alias?

**Comment by rcurtin** [comment]
It should be sufficient to use `memptr()` for each and make sure they are different. :+1:

File: src/mlpack/methods/ann/layer/identity.hpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+#include <mlpack/prereqs.hpp>
+
+#include "layer.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Implementation of the Identity layer.
```
[the remark this answers is not in the document]
Comment (shubham1206agra) [reply src/mlpack/methods/ann/layer/identity.hpp:23]: ?

[the remark this answers is not in the document]
Comment (rcurtin) [reply src/mlpack/methods/ann/layer/identity.hpp:23]: I don't understand what the question is?

[the remark this answers is not in the document]
Comment (shubham1206agra) [reply src/mlpack/methods/ann/layer/identity.hpp:23]: What should I do here?

[the remark this answers is not in the document]
Comment (rcurtin) [reply src/mlpack/methods/ann/layer/identity.hpp:23]: You should add documentation on why you would use this layer, at the very least.  For the comment about the LSTM, let's see what @zoq thinks, and if we do not include any documentation about that, I think it's ok. :+1:

**Comment by shubham1206agra** [comment]
I think we can merge this.

**Review by rcurtin — APPROVED** [review]
Looks great; sorry it took me a little while to get back to this. :)

**Pull request merged**